core/shared/src/main/scala/zio/IO.scala

```diff
@@ -1030,6 +1030,12 @@ object IO {
   def not[E](effect: IO[E, Boolean]): IO[E, Boolean] =
     ZIO.not(effect)
 
+  /**
+   *  @see See [[zio.ZIO.onPlatform]]
+   */
+  def onPlatform[E, A](platform: => Platform)(io: => IO[E, A]): IO[E, A] =
```

Comment:
This raises the question of whether to rename `lock` to `onExecutor`.

Reply:
Yes. We already renamed `on` to `onExecutionContext` so it would definitely be consistent.

Reply:
Will do this in a separate PR.